.circleci/config.yml

```
@@ -0,0 +1,70 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine.
+# See: https://circleci.com/docs/2.0/configuration-reference
+version: 2.1
+orbs:
+  jq: circleci/jq@2.2.0
+
+description: |
+  Run end-to-end integration tests between deno-slack-sdk, Slack Platform CLI and other Slack app development tooling.
+
+jobs:
+  e2e-test:
+    docker:
+      - image: cimg/base:current
+    parameters:
+      e2e_target_branch:
+        type: string
+        description: "What branch of the integration test repo should be used to test the CLI?"
+        # TODO: update to `main` before merge and once https://github.com/slackapi/platform-devxp-test/pull/147 is merged
+        default: deno-sdk-parameter
+      sdk_branch:
+        type: string
+        description: "What deno-slack-sdk branch should be used during testing?"
+        default: main
+    steps:
+      - checkout
+      - run:
+          name: Kick off platform-devpx-test pipeline
+          command: |
+            IMPORT_URL="https://raw.githubusercontent.com/slackapi/deno-slack-sdk/refs/heads/<< parameters.sdk_branch >>/src/";
+            # https://app.circleci.com/settings/organization/github/slackapi/contexts
+            TEST_JOB_WORKFLOW_ID=$(curl --location --request POST 'https://circleci.com/api/v2/project/gh/slackapi/platform-devxp-test/pipeline' \
+            --header 'Content-Type: application/json' \
+            -u "${FIL_CIRCLECI_PERSONAL_TOKEN}:" \
+            --data "{\"branch\":\"<< parameters.e2e_target_branch >>\",\"parameters\":{\"deno_sdk_import_url\":\"${IMPORT_URL}\"}}" | jq '.id')
+            echo "e2e test workflow started with id: $TEST_JOB_WORKFLOW_ID"
+            echo "export TEST_JOB_WORKFLOW_ID=${TEST_JOB_WORKFLOW_ID}" >> $BASH_ENV
+      - run:
+          name: Wait for platform-devxp-test E2E run to complete
+          command: |
+            E2E_RESULT="{}"
+            E2E_STATUS="running"
+            while [[ $E2E_STATUS != "failed" && $E2E_STATUS != "canceled" && $E2E_STATUS != "success" ]]
+            do
+              sleep 10
```

What is the reasoning behind `10`?

Also adding a [specifier suffix](https://www.cyberciti.biz/faq/linux-unix-sleep-bash-scripting/) can make it clearer for new bash devs
```suggestion
              sleep 10s
```

---

Wow didn't know you could do that! Great suggestion. As for 10 seconds: no specific reason. Our tests take a minimum of 1min to run. We could probably bump this delay up to 30s without a huge impact to us and be slightly nicer to our friends at CircleCI.